1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
   Region A: [10, 15, 12, 8, 14]
   Region B: [18, 20, 16, 22, 25]
   Calculate the mean sales for each region.


In [1]:

region_a_sales = [10, 15, 12, 8, 14]
region_b_sales = [18, 20, 16, 22, 25]


mean_sales_a = sum(region_a_sales) / len(region_a_sales)

mean_sales_b = sum(region_b_sales) / len(region_b_sales)

print("Mean sales for Region A:", mean_sales_a)
print("Mean sales for Region B:", mean_sales_b)


Mean sales for Region A: 11.8
Mean sales for Region B: 20.2


Region A:
Sales: [10, 15, 12, 8, 14]
Mean Sales = (10 + 15 + 12 + 8 + 14) / 5 = 59 / 5 = 11.8

Region B:
Sales: [18, 20, 16, 22, 25]
Mean Sales = (18 + 20 + 16 + 22 + 25) / 5 = 101 / 5 = 20.2

Therefore, the mean sales for Region A is 11.8 and for Region B is 20.2.

2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:
   [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
   Calculate the mode of the survey responses.


In [2]:
from statistics import mode

survey_responses = [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]

mode_value = mode(survey_responses)

print("Mode of the survey responses:", mode_value)


Mode of the survey responses: 4


3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
   Department A: [5000, 6000, 5500, 7000]
   Department B: [4500, 5500, 5800, 6000, 5200]
   Calculate the median salary for each department.


In [4]:
import numpy as np


department_a_salaries = np.array([5000, 6000, 5500, 7000])
department_b_salaries = np.array([4500, 5500, 5800, 6000, 5200])


median_a = np.median(department_a_salaries)

median_b = np.median(department_b_salaries)

print("Median salary for Department A:", median_a)
print("Median salary for Department B:", median_b)


Median salary for Department A: 5750.0
Median salary for Department B: 5500.0


4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:
   [25.5, 24.8, 26.1, 25.3, 24.9]
   Calculate the range of the stock prices.


In [5]:
import numpy as np

stock_prices = np.array([25.5, 24.8, 26.1, 25.3, 24.9])

price_range = np.max(stock_prices) - np.min(stock_prices)

print("Range of the stock prices:", price_range)


Range of the stock prices: 1.3000000000000007


5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:
   Group A: [85, 90, 92, 88, 91]
   Group B: [82, 88, 90, 86, 87]
   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.


In [6]:
from scipy.stats import ttest_ind


group_a_scores = [85, 90, 92, 88, 91]
group_b_scores = [82, 88, 90, 86, 87]


t_statistic, p_value = ttest_ind(group_a_scores, group_b_scores)


print("T-statistic:", t_statistic)
print("P-value:", p_value)


T-statistic: 1.4312528946642733
P-value: 0.19023970239078333


6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
   Sales (in thousands): [25, 30, 28, 20, 26]
   Calculate the correlation coefficient between advertising expenditure and sales.


In [7]:
import numpy as np

advertising_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]


correlation_coefficient = np.corrcoef(advertising_expenditure, sales)[0, 1]


print("Correlation coefficient:", correlation_coefficient)


Correlation coefficient: 0.8757511375750132


7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
   [160, 170, 165, 155, 175, 180, 170]
   Calculate the standard deviation of the heights.


In [8]:
import numpy as np


heights = [160, 170, 165, 155, 175, 180, 170]

std_deviation = np.std(heights)

print("Standard deviation of heights:", std_deviation)


Standard deviation of heights: 7.953949089757174


8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]
   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]
   Perform a linear regression analysis to predict job satisfaction based on employee tenure.


In [9]:
import numpy as np
from scipy.stats import linregress


employee_tenure = np.array([2, 3, 5, 4, 6, 2, 4])
job_satisfaction = np.array([7, 8, 6, 9, 5, 7, 6])

# Perform linear regression analysis
slope, intercept, r_value, p_value, std_err = linregress(employee_tenure, job_satisfaction)


print("Slope:", slope)
print("Intercept:", intercept)
print("R-value (Correlation coefficient):", r_value)
print("P-value:", p_value)
print("Standard error:", std_err)


Slope: -0.4680851063829787
Intercept: 8.595744680851062
R-value (Correlation coefficient): -0.520573376708331
P-value: 0.23095550952297988
Standard error: 0.34333860006713723


9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:
   Medication A: [10, 12, 14, 11, 13]
   Medication B: [15, 17, 16, 14, 18]
   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.


In [10]:
import numpy as np
from scipy.stats import f_oneway

# Recovery times for Medication A and Medication B
medication_a_recovery_times = np.array([10, 12, 14, 11, 13])
medication_b_recovery_times = np.array([15, 17, 16, 14, 18])

# Perform ANOVA
f_statistic, p_value = f_oneway(medication_a_recovery_times, medication_b_recovery_times)


print("F-statistic:", f_statistic)
print("P-value:", p_value)


F-statistic: 16.0
P-value: 0.003949772803445326


10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is

 as follows:
    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
    Calculate the 75th percentile of the feedback ratings.


In [11]:
import numpy as np

# Feedback ratings data
feedback_ratings = np.array([8, 9, 7, 6, 8, 10, 9, 8, 7, 8])

# Calculate the 75th percentile
percentile_75 = np.percentile(feedback_ratings, 75)

print("75th percentile of feedback ratings:", percentile_75)


75th percentile of feedback ratings: 8.75


11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.


In [12]:
import numpy as np
from scipy.stats import ttest_1samp


weights = np.array([10.2, 9.8, 10.0, 10.5, 10.3, 10.1])

# Perform one-sample t-test
t_statistic, p_value = ttest_1samp(weights, 10)


print("T-statistic:", t_statistic)
print("P-value:", p_value)


T-statistic: 1.5126584522688367
P-value: 0.19077595151110102


12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:
    Design A: [100, 120, 110, 90, 95]
    Design B: [80, 85, 90, 95, 100]
    Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.


In [13]:
import numpy as np
from scipy.stats import chi2_contingency


design_a_clicks = np.array([100, 120, 110, 90, 95])
design_b_clicks = np.array([80, 85, 90, 95, 100])

# Calculate total clicks for each design
total_clicks_a = np.sum(design_a_clicks)
total_clicks_b = np.sum(design_b_clicks)

# Create a contingency table
contingency_table = np.array([design_a_clicks, design_b_clicks])

# Perform chi-square test
chi2_statistic, p_value, _, _ = chi2_contingency(contingency_table)


print("Chi-square statistic:", chi2_statistic)
print("P-value:", p_value)


Chi-square statistic: 6.110658166925435
P-value: 0.19103526314060293


13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
    Calculate the 95% confidence interval for the population mean satisfaction score.


In [14]:
import numpy as np
from scipy.stats import t

# Satisfaction scores data
satisfaction_scores = np.array([7, 9, 6, 8, 10, 7, 8, 9, 7, 8])

# Calculate the sample mean and standard error
sample_mean = np.mean(satisfaction_scores)
standard_error = np.std(satisfaction_scores, ddof=1) / np.sqrt(len(satisfaction_scores))

# Set the confidence level and degrees of freedom
confidence_level = 0.95
degrees_of_freedom = len(satisfaction_scores) - 1

# Calculate the confidence interval
confidence_interval = t.interval(confidence_level, degrees_of_freedom, loc=sample_mean, scale=standard_error)

print("95% Confidence Interval:", confidence_interval)


95% Confidence Interval: (7.043561120599888, 8.756438879400113)


14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
    Perform a simple linear regression to predict performance based on temperature.


In [15]:
import numpy as np
from scipy.stats import linregress

# Temperature and performance data
temperature = np.array([20, 22, 23, 19, 21])
performance = np.array([8, 7, 9, 6, 8])

# Perform simple linear regression
slope, intercept, r_value, p_value, std_err = linregress(temperature, performance)


print("Slope:", slope)
print("Intercept:", intercept)
print("R-value (Correlation coefficient):", r_value)
print("P-value:", p_value)
print("Standard error:", std_err)


Slope: 0.5
Intercept: -2.9000000000000004
R-value (Correlation coefficient): 0.6933752452815364
P-value: 0.19417134561205843
Standard error: 0.30000000000000004


15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:
    Group A: [4, 3, 5, 2, 4]
    Group B: [3, 2, 4, 3, 3]
    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.


In [16]:
import numpy as np
from scipy.stats import mannwhitneyu

# Preferences data for Group A and Group B
group_a_preferences = np.array([4, 3, 5, 2, 4])
group_b_preferences = np.array([3, 2, 4, 3, 3])

# Perform Mann-Whitney U test
statistic, p_value = mannwhitneyu(group_a_preferences, group_b_preferences)


print("Mann-Whitney U statistic:", statistic)
print("P-value:", p_value)


Mann-Whitney U statistic: 17.0
P-value: 0.380836480306712


16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:
    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
    Calculate the interquartile range (IQR) of the ages.


In [1]:
import numpy as np
from scipy.stats import iqr


ages = np.array([25, 30, 35, 40, 45, 50, 55, 60, 65, 70])

# Calculate the interquartile range
interquartile_range = iqr(ages)

# Print the interquartile range
print("Interquartile Range (IQR) of ages:", interquartile_range)


Interquartile Range (IQR) of ages: 22.5


17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:
    Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]
    Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]
    Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]
    Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.


In [2]:
import numpy as np
from scipy.stats import kruskal

# Accuracy scores data for each algorithm
algorithm_a_scores = np.array([0.85, 0.80, 0.82, 0.87, 0.83])
algorithm_b_scores = np.array([0.78, 0.82, 0.84, 0.80, 0.79])
algorithm_c_scores = np.array([0.90, 0.88, 0.89, 0.86, 0.87])

# Perform Kruskal-Wallis test
statistic, p_value = kruskal(algorithm_a_scores, algorithm_b_scores, algorithm_c_scores)

print("Kruskal-Wallis statistic:", statistic)
print("P-value:", p_value)


Kruskal-Wallis statistic: 9.696947935368053
P-value: 0.007840333026249539


18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:
    Price (in dollars): [10, 15, 12, 8, 14]
    Sales: [100, 80, 90, 110, 95]
    Perform a simple linear regression to predict

 sales based on price.


In [3]:
import numpy as np
from scipy.stats import linregress

# Price and sales data
price = np.array([10, 15, 12, 8, 14])
sales = np.array([100, 80, 90, 110, 95])

# Perform simple linear regression
slope, intercept, r_value, p_value, std_err = linregress(price, sales)


print("Slope:", slope)
print("Intercept:", intercept)
print("R-value (Correlation coefficient):", r_value)
print("P-value:", p_value)
print("Standard error:", std_err)


Slope: -3.506097560975609
Intercept: 136.3719512195122
R-value (Correlation coefficient): -0.8979991308594848
P-value: 0.03850178234753776
Standard error: 0.9918303504036147


19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
    Calculate the standard error of the mean satisfaction score.


In [1]:
import numpy as np
from scipy.stats import sem

# Satisfaction scores data
satisfaction_scores = np.array([7, 8, 9, 6, 8, 7, 9, 7, 8, 7])

# Calculate the standard error of the mean
standard_error_mean = sem(satisfaction_scores)


print("Standard Error of the Mean:", standard_error_mean)


Standard Error of the Mean: 0.30550504633038933


20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
    Sales (in thousands): [25, 30, 28, 20, 26]
    Perform a multiple regression analysis to predict sales based on advertising expenditure.



In [2]:
import numpy as np
import statsmodels.api as sm


advertising_expenditure = np.array([10, 15, 12, 8, 14])
sales = np.array([25, 30, 28, 20, 26])

# Add a constant term to the advertising expenditure data
X = sm.add_constant(advertising_expenditure)

# Perform multiple regression analysis
model = sm.OLS(sales, X)
results = model.fit()


print(results.summary())


                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.767
Model:                            OLS   Adj. R-squared:                  0.689
Method:                 Least Squares   F-statistic:                     9.872
Date:                Tue, 11 Jul 2023   Prob (F-statistic):             0.0516
Time:                        17:04:22   Log-Likelihood:                -9.5288
No. Observations:                   5   AIC:                             23.06
Df Residuals:                       3   BIC:                             22.28
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         12.2012      4.429      2.755      0.0

/opt/conda/lib/python3.10/site-packages/statsmodels/stats/stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
